lib/Caxy/HtmlDiff/Table/TableDiff.php

```diff
@@ -80,11 +82,32 @@ public static function create($oldText, $newText, HtmlDiffConfig $config = null)
      * @param array|null $specialCaseTags
      * @param bool|null  $groupDiffs
      */
-    public function __construct($oldText, $newText, $encoding = 'UTF-8', $specialCaseTags = null, $groupDiffs = null)
+    public function __construct(
+        $oldText,
+        $newText,
+        $encoding = 'UTF-8',
+        $specialCaseTags = null,
+        $groupDiffs = null,
+        $defaultPurifierSerializerCache = null
+    )
     {
         parent::__construct($oldText, $newText, $encoding, $specialCaseTags, $groupDiffs);
 
-        $this->purifier = new \HTMLPurifier(\HTMLPurifier_Config::createDefault());
+        $this->initPurifier($defaultPurifierSerializerCache);
+    }
+
+    /**
+     * Initializes HTMLPurifier with cache location
+     * @param  null|string $defaultPurifierSerializerCache
+     * @return void
+     */
+    protected function initPurifier($defaultPurifierSerializerCache)
+    {
+        $HTMLPurifierConfig = \HTMLPurifier_Config::createDefault();
+        // Cache for the serializer defaults to inside the HTMLPurifier library
+        // under the DefinitionCache/Serializer folder.
+        $HTMLPurifierConfig->set('Cache.SerializerPath', $defaultPurifierSerializerCache);
```

Comment:
Should we only set this if `$defaultPurifierSerializerCache` is not null? Or would there be a situation where one would want to set it to null? (does it just not cache if it's null? or does it just use the default?)